File: src/main/java/kg/mega/kindergarten/services/TeacherService.java
package kg.mega.kindergarten.services;

import kg.mega.kindergarten.enums.Delete;
import kg.mega.kindergarten.enums.Position;
import kg.mega.kindergarten.models.Teacher;
import kg.mega.kindergarten.models.dtos.TeacherSaveDto;
import kg.mega.kindergarten.models.dtos.TeacherDto;

import java.util.List;

public interface TeacherService {
    TeacherDto create(TeacherSaveDto teacherCreateDto, Position position);

    TeacherDto update(Long id, TeacherSaveDto teacherSaveDto,Position position, Delete delete);

    TeacherDto delete(Long id);

    List<Teacher> findAllList(int page, int size);

    Teacher findById(Long id);
    Teacher findByIdForGroup(Long id, Position position);

    Object findGroupByTeacherId(Long teacherId);

}

File: src/api/teacherService.js
import axiosClient from './axiosClient';

const teacherService = {
  getAll: (page = 0, size = 10) => axiosClient.get('/teacher/get-list', { params: { page, size } }),
  getById: (id) => axiosClient.get('/teacher/find-by-id', { params: { id } }),
  create: (teacherData, position) => axiosClient.post('/teacher/create', teacherData, { params: { position } }),
  update: (id, teacherData, position, deleteStatus = 'ACTIVE') => axiosClient.put('/teacher/update', teacherData, { params: { id, position, delete: deleteStatus } }),
  delete: (id) => axiosClient.delete('/teacher/delete', { params: { id } }),
  getGroup: (teacherId) => axiosClient.get('/teacher/group', { params: { teacherId } }),
};

export default teacherService;
